src/vs/workbench/browser/parts/editor/textEditor.ts

```diff
@@ -200,6 +201,18 @@ export abstract class BaseTextEditor extends EditorPane implements ITextEditorPa
 		// Subclasses can override
 	}
 
+	getInternalContextKeyService(): IContextKeyService | undefined {
+		return this.invokeWithinContext(accessor => accessor.get(IContextKeyService));
+	}
+
+	private invokeWithinContext<T>(fn: (accessor: ServicesAccessor) => T): T | undefined {
```

Comment:
I would merge this private method into the `getInternalContextKeyService` method.